### PR TITLE
Stub out future sync-ified API

### PR DESCRIPTION
**Pull request by aubrika**

### CODE REVIEW

File: src/iterator.rs [hunk 1]
```diff
@@ -271,67 +279,116 @@ where
...
 }
 
-impl<R> Iterator for JellyfishMerkleIterator<R>
+impl<R> futures::Stream for JellyfishMerkleStream<R>
 where
-    R: TreeReader,
+    R: TreeReader + Send + Sync + 'static,
 {
-    type Item = Result<(KeyHash, OwnedValue)>;
+    type Item = Result<(KeyHash, ValueHash)>;
```
Comment (hdevalence) [inline src/iterator.rs:344]: Why does this return a `ValueHash` instead of the full `OwnedValue` now?

File: src/tree.rs [hunk 2]
```diff
@@ -26,6 +28,802 @@ pub struct JellyfishMerkleTree<'a, R> {
...
+            // When all children of an internal node are leaves, the leaf count is accessible
+            // even if the migration haven't started. In fact, in such a case, there's no difference
+            // in the old and new serialization format. Forcing it None here just to make the tests
+            // straightforward.
+            Ok(None)
+        }
+    }
+}
+
+/*
```
Comment (hdevalence) [inline src/tree.rs:826]: Looks like the old implementation was commented out but not replaced, could we remove it so that we get the diff of old vs new?

Comment (aubrika) [reply src/tree.rs]: Yep, this was a miss & has been done

File: src/iterator.rs (hunk 1 above)
Comment (aubrika) [reply src/iterator.rs]: Oops -  fixed!

**Pull request merged**